test/integration/component/maint/test_vpc.py

```diff
@@ -0,0 +1,360 @@
+            to "Vmxnet3"
+            #2.Create VPC
+            #3.Create one tier
+            #4.Deploy one guest vm in the tier created in step3
+        """
+        if self.hypervisor.lower() not in ['vmware']:
+            self.skipTest("This test can only run on vmware setup")
+
+        nic_types = Configurations.list(
+            self.apiclient,
+            name="vmware.systemvm.nic.device.type"
+        )
+        self.assertEqual(validateList(nic_types)[0], PASS, "Invalid list config")
+        nic_type = nic_types[0].value
+        reset = False
+        if nic_type.lower() != "vmxnet3":
+            self.updateConfigurAndRestart("vmware.systemvm.nic.device.type", "Vmxnet3")
+            reset = True
+
+        self.services["vpc"]["cidr"] = "10.1.1.1/16"
+        self.debug("creating a VPC network in the account: %s" %
+                   self.account.name)
+        try:
+            vpc = VPC.create(
+                self.apiclient,
+                self.services["vpc"],
+                vpcofferingid=self.vpc_off.id,
+                zoneid=self.zone.id,
+                account=self.account.name,
+                domainid=self.account.domainid
+            )
+            vpc_res = VPC.list(self.apiclient, id=vpc.id)
+            self.assertEqual(validateList(vpc_res)[0], PASS, "Invalid response from listvpc")
+
+            self.network_offering = NetworkOffering.create(
+                self.apiclient,
+                self.services["network_offering"],
+                conservemode=False
+            )
+            # Enable Network offering
+            self.network_offering.update(self.apiclient, state='Enabled')
+            self.cleanup.append(self.network_offering)
+
+            gateway = vpc.cidr.split('/')[0]
+            # Split the cidr to retrieve gateway
+            # for eg. cidr = 10.0.0.1/24
+            # Gateway = 10.0.0.1
+            # Creating network using the network offering created
+            self.debug("Creating network with network offering: %s" %
+                       self.network_offering.id)
+            network = Network.create(
+                self.apiclient,
+                self.services["network"],
+                accountid=self.account.name,
+                domainid=self.account.domainid,
+                networkofferingid=self.network_offering.id,
+                zoneid=self.zone.id,
+                gateway=gateway,
+                vpcid=vpc.id
+            )
+            self.debug("Created network with ID: %s" % network.id)
+            VirtualMachine.create(
+                self.apiclient,
+                self.services["virtual_machine"],
+                accountid=self.account.name,
+                domainid=self.account.domainid,
+                serviceofferingid=self.service_offering.id,
+                networkids=[str(network.id)]
+            )
+            self.debug("Deployed VM in network: %s" % network.id)
+        except Exception as e:
```

Comment:
why rely on the valid implementation by the developer of the error handling in CloudStack? shouldn't you assert that it is actually created? A programmer might have silently ignored an exception.